### PR TITLE
Min API unit and integration testing

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Contributes to #26352
[Internal review URL](https://review.learn.microsoft.com/en-us/aspnet/core/fundamentals/minimal-apis/test-min-api?view=aspnetcore-7.0&branch=pr-en-us-27116)
[test project](https://github.com/dotnet/AspNetCore.Docs.Samples/tree/main/fundamentals/minimal-apis/samples/MinApiTestsSample)

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@javiercn please review [test sample code](https://github.com/dotnet/AspNetCore.Docs.Samples/tree/main/fundamentals/minimal-apis/samples/MinApiTestsSample)

I'm guessing the authorization code won't be approved by @blowdart  so I'll remove that.

**Comment by blowdart** [comment]
You know the TestApplicationWebFactory wasn't a bad way to go about it. ASP.NET has a much fuller featured way to do it in Microsoft.AspNetCore.TestHost which might be more familiar to folks who have gone source spelunking. But it's also overkill for this :)

File: aspnetcore/fundamentals/minimal-apis/test-min-api.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+By [Fiyaz Bin Hasan](https://github.com/fiyazbinhasan), and [Rick Anderson](https://twitter.com/RickAndMSFT)
+
+[!INCLUDE[](~/includes/integrationTests.md)]
+
+The [sample code on GitHub](https://github.com/dotnet/AspNetCore.Docs.Samples/tree/main/fundamentals/minimal-apis/samples/MinApiTestsSample) provides an example of unit and integration tests on a Minimal API app.
+
+## Additional Resources
+
+* [Basic authentication tests](https://github.com/blowdart/idunno.Authentication/tree/dev/test/idunno.Authentication.Basic.Test) is not a .NET repository but was written by a member of the .NET team. It provides examples of basic authentication testing.
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/minimal-apis/test-min-api.md:20]: @blowdart  suggest better wording.

**Comment by Rick-Anderson** [comment]
@rafikiassumani-msft  can you assign a reviewer?

**Comment by Rick-Anderson** [comment]
@maartenba please review the [test project](https://github.com/dotnet/AspNetCore.Docs.Samples/tree/main/fundamentals/minimal-apis/samples/MinApiTestsSample)

**Comment by maartenba** [comment]
Did a quick run through, looks good imo

**Comment by martincostello** [comment]
LGTM too

**Pull request merged**